softhddevice.cpp

```diff
@@ -290,16 +260,38 @@ cSpuDecoder *cSoftHdDevice::GetSpuDecoder(void)
  */
 bool cSoftHdDevice::HasDecoder(void) const
 {
-	return true;
+	std::lock_guard<std::mutex> lock(m_mutex);
+//	LOGDEBUG("device: %s: %d", __FUNCTION__, !m_skipstream);
+
+	return !m_skipstream;
 }
 
 /**
  * Returns true if this device can currently start a replay session
  */
 bool cSoftHdDevice::CanReplay(void) const
+{
+	std::lock_guard<std::mutex> lock(m_mutex);
+	LOGDEBUG("device: %s: %d", __FUNCTION__, !m_skipstream);
+
+	return !m_skipstream;
+}
+
+/**
+ * Stop replaying a recording
+ */
+void cSoftHdDevice::StopReplay(void)
 {
 	LOGDEBUG("device: %s:", __FUNCTION__);
-	return true;
+	cDevice::StopReplay();
+}
+
+/**
+ * Returns true, if the device is replaying
+ */
+bool cSoftHdDevice::Replaying(void) const
+{
+	return cDevice::Replaying();
```

Comment:
These could entirely be removed. IMHO, the debug message could be re-added if needed.